src/Lecture1.hs

```diff
@@ -123,4 +150,15 @@ and lower than 6 elements (4, 5, 6, 7, 8 and 9).
 
 🕯 HINT: Use recursion to implement this function.
 -}
-lowerAndGreater n list = error "TODO"
+lowerAndGreater :: Int ->  [Int] -> String
+lowerAndGreater n list =
+    let 
+        values = lowerHigher n list
+    in
+    show n ++ " is greater than " ++ show (fst values)  ++ " elements and lower than " ++ show (snd values) ++ " elements"
+
+lowerHigher :: Int-> [Int]-> (Int, Int)
+lowerHigher val = foldl lowerGreater (0, 0) 
+    where lowerGreater (l, h) x | x < val = (l+1, h) 
+          lowerGreater (l, h) x | x > val = (l, h+1) 
+          lowerGreater (l, h) _ = (l, h)
```

Comment:
That's a clever solution 😅 🧠 
I haven't explained folds yet so you're already two steps ahead 🦶🏻 🦶🏻 

In general, `foldl` is not recommended and it's better to avoid this particular function. But I believe it's not a problem for the first homework 🙂

Reply:
I was wondering, what is the problem with foldl in this case and what a better solution is for this particular problem.

I first tried to solve it with two functions. Where in the first function I went trough the whole list and returning everything that is lower. And do the same for everything that is higher. But I did not like this, because I had to write two functions, which basically are doing the same and  I had to go through the list twice.

Reply:
`foldl` introduces space leaks. It doesn't evaluate `l+1` and `h+1` immediately so this function will contain unnecessary higher memory usage.

You can write a single function that takes two separate arguments (the number of higher and lower elements) and implement it recursively similar to the `count` function from the presentation. Currently, even this recursive implementation will be lazier than you want as I haven't explained strict evaluation yet. But at least you can easily improve the performance of a manually recursive function. With `foldl` it's not even possible no matter what you try 🙂 
There's `foldl'` function which is much better than `foldl`. So you should use `foldl'` always instead. However, even in this case, if your accumulator is a tuple, you need to take extra care to avoid space leaks.